### PR TITLE
Write a test for safe_unlink

### PR DESCRIPTION
**Pull request by hexagonrecursion**
I am going to backport #6301, but first I'll gift you a test for it.

### CODE REVIEW

**Comment by hexagonrecursion** [comment]
I have to say it: the top two reasons I :heart: borg:
1. When tests fail it is usually because **I** broke something rather than a flaky test or something that got merged even though it broke the tests.
2. :moneybag:

File: src/borg/testsuite/helpers.py [hunk 1]
```diff
@@ -1133,3 +1134,15 @@ def test_iter_separated():
...
+def test_safe_unlink_is_safe(tmpdir):
+    contents = b"Hello, world\n"
+    victim = tmpdir / 'victim'
+    victim.write_binary(contents)
+    hard_link = tmpdir / 'hardlink'
+    hard_link.mklinkto(victim)
+
+    safe_unlink(hard_link)
+
+    assert victim.read_binary() == contents
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/helpers.py:1148]: well, it is ok that we have a test, but we also must face that this only tests the usual behaviour (which is the rather not-so-complicated part of `safe_unlink` and basically just a call to `os.unlink`) and not the more complicated part of it.

Comment (ThomasWaldmann) [reply src/borg/testsuite/helpers.py:1148]: hmm, can you check whether mocking `os.unlink` to raise ENOSPC works?

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**